### PR TITLE
Implement option for Clockwise Initiative

### PR DESCRIPTION
**Pull request by chrpow**
This creates a setting to enable "clockwise initiative" as described on page 83 of the core rulebook. Rather than all characters acting in descending order of their initiative rolls, the combatant with the highest roll goes first, then all other combatants follow in a fixed order (as though they were sitting around a table, going clockwise). The order is determined by the order of users in the User Configuration menu. All NPCs will act on the "GM's turn," going in descending order of initiative bonus.

By default, this option is disabled, as players will likely expect initiative to work the way it does in other Foundry systems, even though technically this is the way Shadowdark is played RAW.

In order to implement this setting, I had to replace the core Combat class with a new EncounterSD class that extends Combat. I also had to add a few strings to the translation files, like name/description for the setting, and some dialog flavor that isn't supported by the base system.

### CODE REVIEW

**Comment by chrpow** [comment]
This sort of solves the original issue I had raised in #590. I still plan to look into rearrangeable initiative, but in the meantime this fixes the root issue of Foundry's initiative methods not being compatible with Shadowdark's ruleset.

**Review by Muttley — CHANGES_REQUESTED** [review]

File: system/src/documents/EncounterSD.mjs
[the remark this answers is not in the document]
Comment (chrpow) [reply system/src/documents/EncounterSD.mjs:1]: I apologize, I forgot I had disabled the auto lint in VS. It should be good now.

**Review by Muttley — APPROVED** [review]

**Comment by Muttley** [comment]
I'm afraid I had to throw away this merge, as the new combat tracker was non-functional and only rolled for NPCs, completely ignoring standard characters.

Unfortunately I don't have time to try and fix it myself currently, so you'll have to revisit it and try again once you've resolved the issue.

**Comment by chrpow** [comment]
> I'm afraid I had to throw away this merge, as the new combat tracker was non-functional and only rolled for NPCs, completely ignoring standard characters.
> 
> Unfortunately I don't have time to try and fix it myself currently, so you'll have to revisit it and try again once you've resolved the issue.

When you were testing, did the PCs you selected not have an associated player? The clockwise initiative tracker can only sort PCs if it knows which Foundry user that actor is associated with.

I just did a quick test: As you can see, only Creeg and Elbin had their initiatives counted since they're the only PC tokens that have an associated player. The NPCs are obviously always rolled because they're associated with game.users.activeGM.
image
If you wanted this tracker to work without assigning characters, I suppose I could just make it sort the PCs by uuid? That would at least keep the order consistent, even though it wouldn't be immediately obvious to the users how the order was generated, and the gamemaster wouldn't be able to rearrange the user order.

Let me know which path I should go down, it should be simple to code either way.

**Pull request merged**